src/odemis/gui/comp/viewport.py

```diff
@@ -42,12 +42,35 @@
 from odemis.gui.util import call_in_wx_main
 from odemis.gui.util.raster import rasterize_line
 from odemis.model import NotApplicableError
-from odemis.util import units, spectrum, peak, img
-import wx
 from odemis.util import no_conflict
-from odemis.acq.stream import PixelTemporalSpectrumProjection, \
-    DataProjection, LineSpectrumProjection
-from odemis.acq.stream._projection import DataProjection
+from odemis.util import units, spectrum, peak
+import wx
+
+
+def get_original_stream(s_or_p):
+    """
+    To support Streams which provide projection on their own, and Streams which
+    rely on a separate projection, find the original Stream.
```

Comment:
a projection or projections?  

> rely on a separate projection. This method finds the original Stream.

What is the original stream?

Reply:
Now we have two layers:
 * at the low-level, it's the "original Stream", which contains the data, and some settings common for all the projections/views.
 * at the higher level, it's the "projection", which has a link/pointer, to the original stream, and provides a `.image` to provide the projected data.

So this function, when there is such double layer, gets the projection (a single one), and returns the "original stream". I'm also not convinced by this naming, but I haven't had better idea. Do you?